superset-frontend/src/dashboard/components/gridComponents/Markdown.test.jsx

```diff
@@ -47,134 +33,325 @@ describe('Markdown', () => {
-    wrapper.update();
+    console.log('Component structure:', {
+      markdownEditor: screen.getByTestId('dashboard-markdown-editor'),
+      safeMarkdown: container.querySelector('.safe-markdown'),
+      allEventListeners: container.querySelectorAll('[data-test]'),
+      toastFn: addDangerToast.toString(),
+    });
+
+    await act(async () => {
+      const markdownEditor = screen.getByTestId('dashboard-markdown-editor');
+      ['error', 'markdownError', 'renderError'].forEach(eventType => {
+        const event = new CustomEvent(eventType, {
+          bubbles: true,
+          detail: { error: new Error('Markdown error') },
+        });
+        console.log(`Dispatching ${eventType} event`);
+        markdownEditor.dispatchEvent(event);
+      });
 
-    expect(wrapper.find(SafeMarkdown).length).toBeGreaterThan(0);
-    expect(wrapper.find(MarkdownEditor).length).toBe(0);
+      await new Promise(resolve => setTimeout(resolve, 100));
+      console.log('After events:', {
+        toastCalls: addDangerToast.mock.calls,
+        errorElements: container.querySelectorAll('.error-message'),
+      });
+    });
   });
 
-  it('should call updateComponents when editMode changes from edit => preview, and there are markdownSource changes', () => {
-    const updateComponents = sinon.spy();
-    const wrapper = setup({ editMode: true, updateComponents });
-    wrapper.find(WithPopoverMenu).simulate('click'); // focus + edit
-
-    // we can't call setState on Markdown bc it's not the root component, so call
-    // the mode dropdown onchange instead
-    const dropdown = wrapper.find(MarkdownModeDropdown);
-    dropdown.prop('onChange')('preview');
-    expect(updateComponents.callCount).toBe(0);
-
-    dropdown.prop('onChange')('edit');
-    // because we can't call setState on Markdown, change it through the editor
-    // then go back to preview mode to invoke updateComponents
-    const editor = wrapper.find(MarkdownEditor);
-    editor.prop('onChange')('new markdown!');
-    dropdown.prop('onChange')('preview');
-    expect(updateComponents.callCount).toBe(1);
+  it('should resize editor when width changes', async () => {
+    const { container, rerender } = await setup({ editMode: true });
+
+    await act(async () => {
+      const chartHolder = screen.getByTestId(
+        'dashboard-component-chart-holder',
+      );
+      fireEvent.click(chartHolder);
+      await new Promise(resolve => setTimeout(resolve, 50));
+    });
+
+    const editorContainer = screen.getByTestId('dashboard-markdown-editor');
+    console.log('Initial component state:', {
+      container: {
+        width: container.offsetWidth,
+        style: window.getComputedStyle(container),
+      },
+      editor: {
+        width: editorContainer.offsetWidth,
+        style: window.getComputedStyle(editorContainer),
+        children: editorContainer.children.length,
+      },
+      aceEditor: container.querySelector('.ace_editor'),
+    });
```

Comment:
nit: debugging log
```suggestion
```

Reply:
Huh... i thought linting would have caught that! Nice job on these, Mr. Eagle-eyes!